src/anyio/_backends/_asyncio.py

```diff
@@ -1857,7 +1875,9 @@ def _cancel_all_tasks(self) -> None:
             return
 
         for task in to_cancel:
-            task.cancel()
+            # Workaround for async generator shutdown on Python 3.6
+            if not isinstance(get_coro(task), async_generator_athrow):
+                task.cancel()
```

Comment:
```suggestion
            # Workaround for async generator shutdown on Python 3.6
            if (
                sys.version_info >= (3, 6)
                and not isinstance(get_coro(task), async_generator_athrow)
            ):
                task.cancel()
```
you do want to cancel athrow tasks on 3.7+

Reply:
If we want to do that on 3.7.5+, wouldn't we be missing such tasks on earlier versions with the proposed code?

Reply:
missing?

Reply:
Not cancelling them even though we possibly should.

Reply:
yes you have to wait an event loop turn to catch those missing tasks on 3.7.4-